File: 盛最多水的容器.cpp
class Solution {
public:
    int maxArea(vector<int>& a) {
        int left = 0, right = a.size() - 1, ans = 0;
        while (left <= right) {
            ans = max(ans, min(a[left], a[right]) * (right - left));
            if (a[left] < a[right]) {
                ++left;
            } else {
                --right;
            }
        }
        return ans;
    }
};

File: 盛最多水的容器.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        n = len(height)
        left, right, ans = 0, n - 1, 0
        while left <= right:
            ans = max(ans, min(height[left], height[right]) * (right - left))
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        return ans
